Server/Validator.cs

```diff
@@ -0,0 +1,76 @@
+﻿using CommonLibrary.CardsClasses;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Server
+{
+	class Validator
+	{
+		public bool ValidateActionCardUsing(ActionCard actionCard, List<ActionCard> tableActionCards)
+		{
+
+			throw new NotImplementedException();
+		}
+
+		public bool ValidateBuildingTunnelAction(RouteCard routeCard, List<RouteCard> tableRouteCards)
+		{
+			//Получаем координаты, куда карточку из руки хотят положить
+			int x = routeCard.Coordinates.Coordinate_X;
+			int y = routeCard.Coordinates.Coordinate_Y;
+
+			//Определяем какие карточки находятся вокруг
+			RouteCard topCardFromTable = tableRouteCards.FirstOrDefault(topCard => topCard.Coordinates.Coordinate_X == x && topCard.Coordinates.Coordinate_Y == y + 1);
+			RouteCard bottomCardFromTable = tableRouteCards.FirstOrDefault(bottomCard => bottomCard.Coordinates.Coordinate_X == x && bottomCard.Coordinates.Coordinate_Y == y - 1);
+			RouteCard leftCardFromTable = tableRouteCards.FirstOrDefault(leftCard => leftCard.Coordinates.Coordinate_X == x - 1 && leftCard.Coordinates.Coordinate_Y == y);
+			RouteCard rightCardFromTable = tableRouteCards.FirstOrDefault(rightCard => rightCard.Coordinates.Coordinate_X == x + 1 && rightCard.Coordinates.Coordinate_Y == y);
+
+			//Создаем переменные валидаторы для соединений
+			bool validateTopJoining = false, 
+				validateBottomJoining = false, 
+				validateRightJoining = false, 
+				validateLeftJoining = false;
+
+			//Проверяем каждое соединение
+			if (routeCard.TopJoining == topCardFromTable.BottomJoining)
```

Comment:
You can do it easier:
bool validateTopJoining = (routeCard.TopJoining == topCardFromTable.BottomJoining);
Please do the same below

Reply:
What did you mean?